bpf/proxy/syncer.go

```diff
@@ -876,86 +941,80 @@ func (s *Syncer) newSvcID() uint32 {
 	return id
 }
 
-func (s *Syncer) matchBpfSvc(bsvc nat.FrontendKey, svcs k8sp.ServiceMap) *svcKey {
-	for svc, info := range svcs {
-		if bsvc.Proto() != ProtoV1ToIntPanic(info.Protocol()) {
-			continue
-		}
-
-		matchNP := func() *svcKey {
-			if bsvc.Port() == uint16(info.NodePort()) {
-				for _, nip := range s.nodePortIPs {
-					if bsvc.Addr().String() == nip.String() {
-						skey := &svcKey{
-							sname: svc,
-							extra: getSvcKeyExtra(svcTypeNodePort, nip.String()),
-						}
-						log.Debugf("resolved %s as %s", bsvc, skey)
-						return skey
+func (s *Syncer) matchBpfSvc(bsvc nat.FrontendKey, svc k8sp.ServicePortName, info k8sp.ServicePort) *svcKey {
```

Comment:
Worth a comment explaining the goal of this method, I think.  When does it consider the values to match vs not?  How are node ports and external IPs handled?

I suggest making the naming super-clear wabout what comes from dataplane and what comes from k8s.  Maybe `bpfSvc`, `k8sSvc`, `k8sInfo`?